src/Forms/GridFieldRefreshButton.php

```diff
@@ -169,8 +166,27 @@ public function hasPendingJob()
     public function handleRefresh()
     {
         if (!$this->hasPendingJob()) {
-            $injector = Injector::inst();
-            $injector->get(QueuedJobService::class)->queueJob($injector->create(CheckForUpdatesJob::class));
+            // Queue the job in the immediate queue
+            $job = Injector::inst()->create(CheckForUpdatesJob::class);
+            $jobDescriptorId = $this->getQueuedJobService()->queueJob($job, null, null, QueuedJob::IMMEDIATE);
+            
+            // Check the job descriptor on the queue
+            $jobDescriptor = QueuedJobDescriptor::get()->filter('ID', $jobDescriptorId)->first();
+
+            // If the job is not immediate, change it to immediate and reschedule it to occur immediately
+            if ($jobDescriptor->JobType !== QueuedJob::IMMEDIATE) {
+                $jobDescriptor->JobType = QueuedJob::IMMEDIATE;
+                $jobDescriptor->StartAfter = null;
+                $jobDescriptor->write();
```

Comment:
How much benefit does this if statement add? If the button scheduled immediately then this would only change jobs that have been scheduled outside of the button to be run immediately right? Is this catching that situation and converting that job into a job that can be used for the button?

Reply:
Unfortunately due to the way the job service works it completely ignores what queue you're trying to add a job to if there is a job of that signature already queued. Essentially this will always run because there is always a job already on the queue, and the `StartAfter` value will be some time in the future. The `queueJob` call above is almost useless unless you don't have the rescheduling job for some reason.